### PR TITLE
[Merged by Bors] - refactor(LocalRing/Basic): generalize to non-commutative (semi)rings

### PR DESCRIPTION
**Pull request by javierlcontreras**
Small generalization of some Lemmas in LocalRing.Basic

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by Ruben-VandeVelde — APPROVED** [review]
lgtm with these whitespace changes

File: Mathlib/RingTheory/LocalRing/Basic.lean [hunk 1]
```diff
@@ -31,6 +31,13 @@ theorem of_nonunits_add [Nontrivial R]
...
 
+
+end Semiring
+
+section CommSemiring
+
+variable [CommSemiring R]
+
 /-- A semiring is local if it has a unique maximal ideal. -/
 theorem of_unique_max_ideal (h : ∃! I : Ideal R, I.IsMaximal) : LocalRing R :=
```
Comment (alreadydone) [inline Mathlib/RingTheory/LocalRing/Basic.lean:42]: It's interesting that this holds also for `Ring`s, which inspired me to ask a question: https://math.stackexchange.com/questions/5004383/semiring-with-a-unique-left-ideal

**Comment by YaelDillies** [comment]
I've fixed the conflict. Looks good, thanks!

maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Comment by jcommelin** [comment]
bors d+

**Pull request closed without merging**